posts/2024-05-13-jakarta-nosql-challenge-switching-nosql-easily.adoc

```diff
@@ -0,0 +1,258 @@
+. Go to the "Buckets" section
+. Create a `jnopo` Bucket by clicking on the "Add Bucket" option
++
+[.img_border_light]
+image::/img/blog/couchbase-04.png[Couchbase 04, align="center" width=80%]
+. Click on "Scopes & Collections" and then click on "Add Collection"
+. Enter `GameMatch` as the collection name and click the "Save" button to complete your creation
++
+[.img_border_light]
+image::/img/blog/couchbase-05.png[Couchbase 05, align="center" width=80%]
+. Now, in the "Query" section, run this command https://www.couchbase.com/products/n1ql/[N1QL] to create the primary index for the `GameMatch` collection:
++
+[source, n1ql]
+----
+CREATE PRIMARY INDEX primaryGameMatch ON jnopo._default.GameMatch
+----
++
+[.img_border_light]
+image::/img/blog/couchbase-06.png[Couchbase 06, align="center" width=80%]
+
+With a Couchbase cluster running, we just need to set up Eclipse JNoSQL to connect to it.
+
+Apply the following changes into the `resources/META-INF/microprofile-config.properties` file:
+
+[source, properties]
+----
+## Couchbase
+jnosql.couchbase.host=couchbase://localhost
+jnosql.couchbase.user=root
+jnosql.couchbase.password=123456
+jnosql.document.provider=org.eclipse.jnosql.databases.couchbase.communication.CouchbaseDocumentConfiguration
+----
+
+The `jnosql.document.provider` property is required in this context because we have more than one Eclipse JNoSQL Database API for Document NoSQL implementations. Learn more about these configurations about Eclipse JNoSQL Database API for Couchbase at this https://github.com/eclipse/jnosql-databases/tree/main?tab=readme-ov-file#configuration-2[Git repository].
+
+That's all! Let's restart the https://openliberty.io/[Open Liberty] runtime in order to be able to make sure that everything is running smoothly:
+
+* On Linux/Mac machines, perform the following command:
++
+[source, bash]
+----
+./mvnw liberty:dev
+----
+
+* On Windows machines, perform the following command:
++
+[source, powershell]
+----
+mvnw.cmd liberty:dev
+----
+
+After some game matches, here are the evidences:
+
+[.img_border_light]
+image::/img/blog/couchbase-verification-01.png[Couchbase Verification 01, align="center" width=80%]
+[.img_border_light]
+image::/img/blog/couchbase-verification-02.png[Couchbase Verification 02, align="center" width=80%]
+
+If you're interesting to take a look on the finished project code version developed during this challenge, clone this https://github.com/OpenLiberty/sample-jakartaNoSQL-game[Git repository] and navigate to the `database-switching-challenge` directory.
+
+[source, bash]
+----
+git clone https://github.com/OpenLiberty/sample-jakartaNoSQL-game.git
+cd sample-jakartaNoSQL-game
+cd database-switching-challenge
+----
+
+== Key Takeaways
+
+This challenge was an amazing opportunity to learn how it's ease to switch between NoSQL solutions by using Jakarta NoSQL, bringing one more item to your utility belt that could be interesting to save resources and costs.
```

Comment:
```suggestion
This blog has shown just how easy it can be to switch between NoSQL solutions by using Jakarta NoSQL, bringing one more item to your utility belt that could be useful in helping you save resources and costs.
```